blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/RollupService.java

```diff
@@ -266,8 +266,17 @@ public void run() {
                 final SlotKey slotKey = context.getNextScheduled();
                 if (slotKey == null) { continue; }
                 try {
-                    log.debug("Scheduling slotKey {} @ {}", slotKey, context.getCurrentTimeMillis());
-                    locatorFetchExecutors.execute(new LocatorFetchRunnable(context, slotKey, rollupReadExecutors, rollupWriteExecutors, enumValidatorExecutor));
+                    UpdateStamp stamp = shardStateManager.getUpdateStamp(slotKey);
+                    long currentTimeMillis = context.getCurrentTimeMillis();
+                    final long timeElapsedSinceLastRollup = currentTimeMillis - stamp.getLastRollupTimestamp();
+                    boolean isReroll = timeElapsedSinceLastRollup < ShardStateManager.DELAYED_METRICS_MAX_ALLOWED_DELAY;
```

Comment:
I am not 100% sure of the behavior of premature rollups at this point. I am trying to avoid writing a premature comment.